Build error when building with OpenCL: ‘cout’ is not a member of ‘std’

When I try to build Arrayfire (following the [wiki](https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Linux)) with OpenCL (AMD APP SDK 3.0 or Intel SDK) I get the following error (occurs multiple times in this file):
`In file included from /home/<user>/arrayfire/src/backend/opencl/program.cpp:10:0:
/home/<user>/arrayfire/src/backend/opencl/program.cpp: In function ‘void opencl::buildProgram(cl::Program&, int, const char**, const int*, std::__cxx11::string)’:
/home/<user>/arrayfire/src/backend/opencl/program.hpp:26:18: error: ‘cout’ is not a member of ‘std’
             std::cout << PROG.getBuildInfo<CL_PROGRAM_BUILD_LOG>(       \`

Could be prevented by adding `#include <iostream>` in `src/backend/opencl/program.cpp`, but this is probably not the best way to fix it. Nevertheless build functions normally afterwards.

This error obviously occurs only when enabling OpenCL, the CPU backend works flawless.

OS: openSUSE Tumbleweed 64bit (Kernel 4.13.9-1-default)
ArrayFire: 42a99a76684b662e30c3d76c896069d820ca9a96 (latest master)
cmake: 3.9.4
make: 4.2.1
gcc/g++: 7.2.1

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,7 +109,6 @@ if(NOT LAPACK_FOUND)
     endif()
 endif()
 
-# TODO(umar): Enable other backends
 add_subdirectory(src/backend/common)
 add_subdirectory(src/api/c)
 add_subdirectory(src/api/cpp)

--- a/src/backend/opencl/kernel/histogram.hpp
+++ b/src/backend/opencl/kernel/histogram.hpp
@@ -17,8 +17,12 @@
 #include <Param.hpp>
 #include <debug_opencl.hpp>
 
+using cl::Buffer;
+using cl::EnqueueArgs;
 using cl::Kernel;
 using cl::KernelFunctor;
+using cl::NDRange;
+using cl::Program;
 
 namespace opencl
 {

--- a/src/backend/opencl/kernel/memcopy.hpp
+++ b/src/backend/opencl/kernel/memcopy.hpp
@@ -21,10 +21,12 @@
 #include <debug_opencl.hpp>
 
 using cl::Buffer;
-using cl::Program;
-using cl::KernelFunctor;
 using cl::EnqueueArgs;
+using cl::Kernel;
+using cl::KernelFunctor;
 using cl::NDRange;
+using cl::Program;
+
 using std::string;
 
 namespace opencl

--- a/src/backend/opencl/kernel/nearest_neighbour.hpp
+++ b/src/backend/opencl/kernel/nearest_neighbour.hpp
@@ -18,7 +18,12 @@
 #include <common/dispatch.hpp>
 #include <cache.hpp>
 
+using cl::Buffer;
+using cl::EnqueueArgs;
 using cl::KernelFunctor;
+using cl::Kernel;
+using cl::NDRange;
+using cl::Program;
 using cl::LocalSpaceArg;
 
 namespace opencl

--- a/src/backend/opencl/program.hpp
+++ b/src/backend/opencl/program.hpp
@@ -10,24 +10,17 @@
 #pragma once
 #include <platform.hpp>
 #include <common/util.hpp>
-#include <string>
-#include <mutex>
 
-using cl::Buffer;
-using cl::Program;
-using cl::Kernel;
-using cl::EnqueueArgs;
-using cl::NDRange;
-using std::string;
+#include <cstdio>
+#include <string>
 
 #define SHOW_DEBUG_BUILD_INFO(PROG) do {                                \
         cl_uint numDevices = PROG.getInfo<CL_PROGRAM_NUM_DEVICES>();    \
         for (unsigned int i = 0; i<numDevices; ++i) {                   \
-            std::cout << PROG.getBuildInfo<CL_PROGRAM_BUILD_LOG>(       \
-                PROG.getInfo<CL_PROGRAM_DEVICES>()[i]) << std::endl;    \
-                                                                        \
-            std::cout << PROG.getBuildInfo<CL_PROGRAM_BUILD_OPTIONS>(   \
-                PROG.getInfo<CL_PROGRAM_DEVICES>()[i]) << std::endl;    \
+          printf("%s\n", PROG.getBuildInfo<CL_PROGRAM_BUILD_LOG>(       \
+                      PROG.getInfo<CL_PROGRAM_DEVICES>()[i]).c_str());  \
+          printf("%s\n", PROG.getBuildInfo<CL_PROGRAM_BUILD_OPTIONS>(   \
+                      PROG.getInfo<CL_PROGRAM_DEVICES>()[i]).c_str());  \
         }                                                               \
     } while(0)                                                          \
 

